### ISSUE TITLE
Audio passthrough does not start initially if Adjust Display Refresh Rate is enabled 

### ISSUE THREAD

**Issue by kbuffington** [xbmc/xbmc#15490]
### Describe the bug

In v18, when "Adjust Display Refresh Rate" is selected, audio passthrough does not initially start working and the audio output is PCM. While the movie is playing, the movie can be paused and bringing up the audio menu and see that Audio Passthrough is enabled. While in the movie, disabling passthrough and then re-enabling causes passthrough to work correctly. This worked correctly in Krypton.

If "Adjust Display Refresh Rate" is disabled then bitstreaming starts immediately on playback.

This might be related to using Intel UHD 630 as there are [several reports of users](https://forum.kodi.tv/showthread.php?tid=340358) using this same onboard graphics chipset.

## Expected Behavior

Passthrough should start immediately on playing the video, whether Adjust Display Refresh Rate is enabled or not.

### To Reproduce

1. Set "Allow Passthrough" in the Audio Settings Menu
2. Start playback of a movie with a supported audio format (such as AC-3).
3. See that the output is PCM.
4. Bring up Audio menu of video.
5. Disable Audio Passthrough option.
6. Re-enable Audio Passthrough.
-> Video is now bitstreaming properly.

### Debuglog

The debuglog can be found here: https://paste.kodi.tv/edelivuhoq.kodi

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10
 - Kodi version: v18.0
 - Graphics chipset: Intel UHD 630

**Comment by kbuffington** [comment]
Another User report [here](https://forum.kodi.tv/showthread.php?tid=340569)

**Comment by jjd-uk** [comment]
@afedchin Do you have the time to take I look?

From the log I see audio device as being **WASAPI: DENON-AVRHD (Intel(R) Display Audio)** which has device id **{943EE6CA-43D7-4651-82A6-2028947707B3}** however it appears that device goes missing when it comes to opening the files, as for example:

**L1262** 19:16:43.400 T:15880   DEBUG: CActiveAESink::OpenSink - trying to open device WASAPI:{943EE6CA-43D7-4651-82A6-2028947707B3}

is followed by

**L1265** 19:16:43.401 T:15880   ERROR: CAESinkWASAPI::Initialize: Retrieval of WASAPI endpoint failed. - HRESULT = -2147467259 ErrorMessage = (null)

Thus it falls back to Directsound.

**Comment by afedchin** [comment]
@jjd-uk it looks like a driver issue. As you can see in the log after resolution change there is no `HDMI - DENON-AVRHD (Intel(R) Display Audio)` with '{943EE6CA-43D7-4651-82A6-2028947707B3}' see line 1242, after some time the device appears again (see line 1639) and Kodi opens it (see 1661) with `AE_FMT_S24NE4MSB` format. I don't know why it doesn't open with DTS format because device supports it.

**Comment by afedchin** [comment]
@fritsch do you have an idea why Kodi opens device with `AE_FMT_S24NE4MSB` instead of `DTS`?

**Comment by kbuffington** [comment]
@afedchin I don't think it's just a driver issue as this is a regression from 17.6 where passthrough works fine immediately. This is a log from the same computer with a portable 17.6 Kodi install: https://paste.kodi.tv/farocuruvo.kodi

Reports in the forums were that this started happening around 18 beta 4, and definitely by 18 RC1.

**Comment by kbuffington** [comment]
One other datapoint that will probably just confuse things. I noticed that if I set the output device to "WASAPI: Default" instead of "WASAPI: DENON-AVRHD", then DTS and Dolby Digital bitstreaming start. DTS+HD MA tracks report as DTS on my receiver and not DTS:MA so they seem to be playing the DTS Core track (which apparently still includes height information?). 

Anyway, it's a better-than-nothing workaround, but still not great. For Atmos tracks you always have to toggle on and off the passthrough while the movie is playing otherwise you never get anything from the height speakers.

**Comment by afedchin** [comment]
I see only one difference between v17 and v18 it's different format which the device is opened with after re-initialization. and actually AE part it's a dark forest for me, so I can say nothing more. I hope @fritsch can throw a light to the issue.

**Comment by fritsch** [comment]
From the log WASAPI device gets lost after display lost.

Try to combine it with "Pause before Refreshrate switch" and set it to 15 seconds ... let's then check the log, please. This setting is in the Display Section.

**Comment by kbuffington** [comment]
@fritsch I had set it to a high value before and no dice, but tried again with the max of 20 seconds. Still defaulting to PCM. Log [here](https://paste.kodi.tv/odavuxowuk.kodi).

**Comment by afedchin** [comment]
@fritsch in the log
```
00:38:29.972 T:4868  NOTICE: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder eac3
```
but it should be `CDVDAudioCodecPassthrough`. it occurs before the device is becoming back, so audio codec is selected before device is available

**Comment by afedchin** [comment]
in v17 the codec is selected before refresh rate switch, so it selects right pass-through codec.

**Comment by iz8mbw** [comment]
I have about this problem on my Nvidia Shield (Android).
In my case the audio problem is when changing during playback from a Dolby Digital movie to a DTS movie. In my case, during Dolby Digital playback, if I put in play a DTS movie, Kodi will not go to switch audio track. May be my problem is related to that one.

**Comment by afedchin** [comment]
we are talking about the issue on **Windows** not on Android.

**Comment by iz8mbw** [comment]
And you are sure that problem affect only Windows?

**Comment by afedchin** [comment]
And you are sure that your issue related to the OP? If not please create your own ticket and if a developer decides your will be united with this.

**Comment by iz8mbw** [comment]
Also on my debug log I have lots of `dropping packet type:2 dts:` and `dropping packet type:1 dts:`. Anyway I will open a new ticket. Thanks.

**Comment by kbuffington** [comment]
@afedchin would it be helpful if I tried this with the different RC/Beta candidates to determine when the regression happened?

**Comment by kbuffington** [comment]
Well I did some tests, and bistreaming works on play start in 18beta3 and 18beta4, but _not_ in 18beta5. Not sure if that narrows down the changes where this went badly.

**Comment by FernetMenta** [comment]
This is a scenario that hasn't been considered in the code. In other words it was working by chance in earlier versions. VideoPlayerAudio checks passthrough capabilities at the time it opens a stream. This can happen at every time, not just at start of playback.
The mentioned commit introduced no error, it just brought an already existing issue to surface. A audio device change can happen at every time. VideoPlayerAudio needs to re-evaluate passthrough capabilities after a device change and act accordingly.

**Comment by kbuffington** [comment]
@FernetMenta I agree with this. Audio device changes can cause several issues and is unfortunately the most frustrating aspect of Kodi for me, and probably many others, since it's one of the few things that doesn't "just work". If this leads to fixing that, it'll be a huge benefit to everybody. I'm just hoping someone smarter than me is able to work on it. :)

**Comment by FernetMenta** [comment]
btw. closely related is the sledgehammer method of restarting entire player when passthrough setting is changed: https://github.com/xbmc/xbmc/blob/master/xbmc/Application.cpp#L1045

**Comment by iz8mbw** [comment]
@afedchin opened issue https://github.com/xbmc/xbmc/issues/15523
Thank you

**Comment by iz8mbw** [comment]
@kbuffington did you try if with the last (28 Feb) nightly build you still have that issue?

**Comment by kbuffington** [comment]
I'll give it a try tonight.

**Comment by kbuffington** [comment]
Still having the same issue. Attaching a log of Multichannel starting on playback, then disabling passthrough, and re-enabling to start bit-streaming Dolby Atmos. Not sure if it shows anything different from the original log.

https://paste.kodi.tv/amixunabim.kodi

**Comment by SanderSp** [comment]
I'm having the same problem. Do we know if someone is working on a fix?

**Comment by fritsch** [comment]
We know: currently no one.

**Comment by blacklord049** [comment]
Still happening with 18.5, I was hoping that the issue was solved.

**Comment by HitcherUK** [comment]
> I was hoping that the issue was solved.

Did you read the previous posts?

> Do we know if someone is working on a fix?

> currently no one.

**Comment by maisun** [comment]
Same problem here, very annoying! Hope it can be fixed

**Comment by Indiana999** [comment]
Same problem.  Just upgraded to 4k Projection, new Dell PC with Intel UHD 630 and Marantz SR8012 and experiencing the exact same issue with Passthrough audio.  It started while troubleshooting sluggish video last night that I fixed by setting 'Adjust Display Refresh Rate' to On.  But then immediately had trouble with audio defaulting back to 7.1 from 11.1 when I played any video. Toggling Passthrough during playback corrected audio back to Dolby Atmos.  Hoping it'll be fixed soon.

Love Kodi though. Thank you to all involved.

**Comment by Enominus** [comment]
Same problem! Hope someone can fix it...
Could I help anyhow? Logs, bugreport, etc?

**Comment by DaVukovic** [comment]
Guys, please stop posting "me too"s. That won't help anyone. I guess the dev in question is kind of aware what the problem might be and I also guess he will work on it as soon he has the time to dig into. 

Posting "me too"s just annoys about 12 ppl receiving emails if they don't unsubscribe from this ticket. If they do, this ticket will be forgotten. So just be patient. 

Thanks in advance

**Comment by maisun** [comment]
> Guys, please stop posting "me too"s. That won't help anyone. I guess the dev in question is kind of aware what the problem might be and I also guess he will work on it as soon he has the time to dig into.
> 
> Posting "me too"s just annoys about 12 ppl receiving emails if they don't unsubscribe from this ticket. If they do, this ticket will be forgotten. So just be patient.
> 
> Thanks in advance

I agree with you that "me too" postings won't help to solve the issue.
I think people want to raise it to show it affects more, and hopefully bring some attention from Kodi team.

**Comment by Indiana999** [comment]
This issue has apparently been ongoing for over a year, and not one that I had experienced until I recently spent considerable time, effort and money to upgrade my entertainment system.  The flickering on my 120" screen is rediculous (only in Kodi) and requires the Refresh feature in Kodi to be on.  But doing so now requires me to toggle the 'passthrough' option on each video playback for optimal sound.  Not ideal!

I understand your 'me too' comments are not helpful to you.  But after several hours of frustration and troubleshooting, and finally coming across this article that described my same issue, and it not referencing a solution, and not having been updated in several months, it is nice to hear that it is indeed a known issue that others are still experiencing so that I need not continue to troubleshoot and pull my hair out while doing so.

I have been using Kodi since XMBC on the original XBOX.  I LOVE Kodi and have absolute respect and admiration for the developers.  I have donated on numerous occasions and will continue to do so.

1. Based on the last two posts, I will try to find where and how to post a bug report.  I have never needed to do this before now.  Your response will only be helpful if you can guide me and others in the right direction as to how best to do so.  The more people that report this issue, properly I agree, the more likely it'll be addressed eventually.

2. Based on the post above by kbuffington, I'll attempt to install an earlier version of Kodi that hopefully works as expected with the Refresh and Passthrough options.  Suggestions are welcome.

3. I have tried installing the latest Pre-release and Development Builds and can confirm that the issue still exists in both.

Hopefully, this response will assist others to know that this issue is still existing and unresolved.  Maybe someone can then offer a suggestion as to what they are doing to work around the issue in the meantime.

My respect to you all.

**Comment by captainfearless** [comment]
I have this problem, can anyone post a link to 18beta4 please?, or confirm if any more recent builds solve the issue.

**Comment by maisun** [comment]
No it’s not solved

**Comment by Indiana999** [comment]
I have been using the Kodi-HDR-Edition for a couple months now.  I do not have audio issues with this build, and my 4k videos are in HDR.  The only issue with this build is that the search feature doesn't work as expected.

[https://github.com/fandangos/Kodi-HDR-Edition/releases](url)

Tested and working perfectly.  Highly recommended for those having audio passthrough issues.

A later build, that I have not yet tested, can be found here:  [https://www.xbmcbrasil.net/thread-4527-post-36939.html](url)

**Comment by captainfearless** [comment]
Didn’t work sadly, everything still plays in pcm.

I’m awaiting permission to download the newer one.
Paul Constance


<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by captainfearless** [comment]
Also i now have  a ton of error messages and most addons are saying incompatible and not working, youtube etc. Obviously it's at my own risk, but reverting to the current stable version doesn't fix it, any suggestions would be appreciated!

**Comment by maisun** [comment]
Hi, looks like HDR branch is merged back to the main branch. Does it mean in Kodi 19 this problem will be finally solved?

**Comment by afedchin** [comment]
> Hi, looks like HDR branch is merged back to the main branch. Does it mean in Kodi 19 this problem will be finally solved?

you can try alpha1 to be sure.

**Comment by lynxstrike** [comment]
It happens to me too, and it's independent from "refresh rate".
It's random... some times it makes the change, some times it does not.

It started in Kodi 18, but it keeps happening till now, Kodi Matrix Alpha.

Windows 10 Pro x64
Kodi 19,  KodiSetup-20200807-42869732-test-all-v4.2.0-x64
Nvidia GTX1060, 6GB
( I tried Windows 10 default driver, and all sorts of nvdia drivers... no differance)

**Comment by xen0m0rph2020** [comment]
same here on Kodi 18.8 on LE

**Comment by jhanda1** [comment]
Any progress here?  Kodi 19 Beta 1 here and still having this issue.  17.6 worked perfectly, I skipped the entire 18 series because of this.

**Comment by maisun** [comment]
Looks like the fix is not prioritised. It is a big issue for many Windows users, not sure why it doesn't bring any attention from dev team

**Comment by fuzzard** [comment]
Short answer, there are no active windows Devs.

**Comment by fuzzard** [comment]
Thats a question better asked on the forums.

**Comment by DaveTBlake** [comment]
> why it doesn't bring any attention from dev team

- the active team is very small and the number of dev tasks very big
- devs are volunteers, they individually chose what they work on and when they work on it. 
- there is no external resources to call in to work on anything specific
- this is no overall priorisation of jobs
- only one or two devs run Kodi on Windows devices
- I am one of those, I do not use audio passthrough or have any way to reproduce the issue.
- the player code needs specialist knowledge and experience, this is not currently available.

Reading back "This is a scenario that hasn't been considered in the code. In other words it was working by chance in earlier versions."  hence what is needed is less a fix and more a proper implementation for the first time.

Your best hope is that a member of the wider community with sufficient dev expertise,  that uses Windows and has an interest in getting audio passthough working comes forwards to contribute and do the work. It seems unlikley that a solution would come from the currently active team.

**Comment by captainfearless** [comment]
I‘ve set VLC as my default player, and it solves the problem.

**Comment by Indiana999** [comment]
Folks,

HDR and audio passthrough are now working perfectly for me using Kodi v19 for Windows.  Beta 1 crashed far too often, so I just installed the latest 'nightly' release. This build has also fixed the missing thumbnails in Favorites. I am exceptionally pleased again with Kodi and its developers.  :-)

A couple suggestions:

1. I had major trouble (with audio and HDR) while using a Dell Micro PC (i9 with 16gb RAM) using Intel UHD Graphics 630.  I swapped my micro PC with a tower PC, and installed an NVidia geforce gtx 1080 to resolve most of my audio/video issues.
2. I would strongly recommend a clean install of Kodi while troubleshooting issues.  Rename your data folders and start fresh.
3. For optimal audio settings (Settings / System / Audio - using Expert view), I 'Allow passthrough' and chose WASAPI HDMI as my audio device.
4. For optimal video settings (Settings / Player / Videos - using Exper view), I have presently left all settings at default.
5. The 'speaker setup' in Windows 10 (right-click your volume control next to clock at bottom-right), I have set to 5.1 surround.  I had audio issues with Kodi when I set my Windows speaker setup to Dolby Atmos.

I no longer have any issues with Kodi, and look forward to a final release of Kodi 19.  Again, a huge thanks to the dev team.  You guys are amazing!

**Comment by lynxstrike** [comment]
It's working perfectly for me too since the beginning of Beta 1 (or about).
Many thanks to all developers for their hard work..

**Comment by maisun** [comment]
@Indiana999 @lynxstrike 
The issue is not related to HDR, it's related to auto refresh rate change. It seems that when the refresh rate is changed, Kodi lost connection to AVR and therefore assume there is no passthrough capability of the AVR. Every time I have to disable passthrough and re-enable it in audio settings to bring it back. The issue started in V18 I believe.

**Comment by smvicente** [comment]
Hi! I just wanted to confirm this issue is not resolved as pointed out by the users above, at least when using Intel UHD 630 Graphic card.

I know jogal has done a lot of work on Kodi for Windows, mainly related with HDR but with other bug fixes too, so not sure if he is aware of this as I know he was using both Intel and nVidia, so maybe he is interested and has the technical expertise needed, but I don't know what his user is here, if any...

**Comment by maisun** [comment]
> Hi! I just wanted to confirm this issue is not resolved as pointed out by the users above, at least when using Intel UHD 630 Graphic card on Kodi 19.
> 
> I know jogal has done a lot of work on Kodi for Windows, mainly related with HDR but with other bug fixes too, so not sure if he is aware of this as I know he was using both Intel and nVidia, so maybe he is interested and has the technical expertise needed, but I don't know what his user is here, if any...

Seems that windows is not main stream platform now and an serious issue like this hasn’t been addressed since Kodi 18.

**Comment by maisun** [comment]
Does anybody find a workaround to this? I know I can go to audio setting disable and enable passthrough, then it’ll work. Maybe add a shortcut to toggle passthrough?

**Comment by thexai** [comment]
@maisun is there any possibility that you or someone affected by this problem will publish an updated debug log (Matrix v19.1 / recent nightlie) and explain in some detail what configuration and hardware is in use. I have been read the thread (from end to up) and I do not see that information only "Same problem here".

I myself use AVR Denon + 4K HDR TV and audio passthrough and have no problem with this. I can not do anything if I do not know in what conditions it happens. I do not guarantee that I will fix it but without any recent information sure not.

**Comment by maisun** [comment]
> @maisun is there any possibility that you or someone affected by this problem will publish an updated debug log (Matrix v19.1 / recent nightlie) and explain in some detail what configuration and hardware is in use. I have been read the thread (from end to up) and I do not see that information only "Same problem here".
> 
> I myself use AVR Denon + 4K HDR TV and audio passthrough and have no problem with this. I can not do anything if I do not know in what conditions it happens. I do not guarantee that I will fix it but without any recent information sure not.

Hi @thexai 
Thank you for looking into it! Below I put some description of the problem and my setup, I'm not able to post logs now but will try next week. 
**Problem description:**  Since Kodi 18, I'm not able to get auto-refresh rate change to work together with audio passthrough. In earlier versions it worked well without any issue. When auto-refresh rate change is enabled, Kodi seems to lose the connection to the audio device and somehow won't try to bitstream audio tracks to the device. Once the refresh rate is changed and movie starts to play, Kodi 18 transcode audio tracks to LPCM despite that passthrough is enabled. With Kodi 19 I get no sound. If I go to audio settings (context menu) disable and re-enable passthrough, it will work again.
**My setup:** Lenovo T460P laptop with integrated intel graphics card, AVR is Onkyo, and display is Panasonic Full HD projector. I run Kodi latest stable version from Windows App Store. Not sure what more information you need?

**Comment by thexai** [comment]
> Kodi 18 transcode audio tracks to LPCM
This is not passthrough either

> Windows App Store
Try "normal" Windows desktop x64 version

> T460P laptop
What Windows?  Updated drivers? Intel drivers? Lenovo drivers?

**Comment by captainfearless** [comment]
That’s still clunky, best way is to make VLC default player, then it just works.

**Comment by maisun** [comment]
> > Kodi 18 transcode audio tracks to LPCM
> 
> This is not passthrough either
> 
> > Windows App Store
> 
> Try "normal" Windows desktop x64 version
> 
> > T460P laptop
> 
> What Windows? Updated drivers? Intel drivers? Lenovo drivers?

I’ll try the normal x64 version, but just out of curiously, are you not able to reproduce this? Do you have auto refresh rate enabled and still possible to bitstream dtshd or truehd?

**Comment by maisun** [comment]
> That’s still clunky, best way is to make VLC default player, then it just works.
> […](#)

Why vlc?

**Comment by FernetMenta** [comment]
> @maisun is there any possibility that you or someone affected by this problem will publish an updated debug log (Matrix v19.1 / recent nightlie) and explain in some detail what configuration and hardware is in use. I have been read the thread (from end to up) and I do not see that information only "Same problem here".
> 
> I myself use AVR Denon + 4K HDR TV and audio passthrough and have no problem with this. I can not do anything if I do not know in what conditions it happens. I do not guarantee that I will fix it but without any recent information sure not.

If you want to understand what's going wrong, read my comments. I explained the root cause.

**Comment by maisun** [comment]
> > Kodi 18 transcode audio tracks to LPCM
> 
> This is not passthrough either
> 
> > Windows App Store
> 
> Try "normal" Windows desktop x64 version
> 
> > T460P laptop
> 
> What Windows? Updated drivers? Intel drivers? Lenovo drivers?

Tried the version downloaded from Kodi.tv and same problem.

**Comment by thexai** [comment]
Here is a possible solution, I can't test it because it works well for me anyway, but it looks good:
https://github.com/thexai/xbmc/tree/fix-passthrough

```
2021-05-11 11:49:26.076 T:10784    INFO <general>: Creating audio thread
2021-05-11 11:49:26.126 T:5416     INFO <general>: running thread: CVideoPlayerAudio::Process()
2021-05-11 11:49:26.126 T:10784    INFO <general>: Opening stream: 2 source: 256
2021-05-11 11:49:26.148 T:5416     INFO <general>: CAEStreamParser::SyncDTS - dtsHD MA stream detected (8 channels, 48000Hz, 16bit BE, period: 8192, syncword: 0x41a29547, target rate: 0x18, framesize 4336))
2021-05-11 11:49:26.148 T:5416  WARNING <general>: CVideoPlayerAudio: Display reset has occurred, checking for passthrough
2021-05-11 11:49:26.148 T:5416     INFO <general>: Creating audio stream (codec id: 86020, channels: 8, sample rate: 48000, pass-through)
2021-05-11 11:49:26.160 T:6016     INFO <general>: CAESinkWASAPI::InitializeExclusive: WASAPI Exclusive Mode Sink Initialized using: AE_FMT_S24NE4MSB, 44100, 2
2021-05-11 11:49:26.212 T:6016     INFO <general>: CActiveAESink::OpenSink - initialize sink
2021-05-11 11:49:26.271 T:6016     INFO <general>: CAESinkWASAPI::InitializeExclusive: Format is Supported - will attempt to Initialize
2021-05-11 11:49:26.276 T:8304     INFO <general>: CDVDVideoCodecFFmpeg::Open() Using codec: HEVC (High Efficiency Video Coding)
2021-05-11 11:49:26.285 T:6016     INFO <general>: CAESinkWASAPI::InitializeExclusive: WASAPI Exclusive Mode Sink Initialized using: AE_FMT_S16NE, 192000, 8
2021-05-11 11:49:26.467 T:3304     INFO <general>: CWIN32Util::ToggleWindowsHDR: Toggle Windows HDR On (OFF => ON).
2021-05-11 11:49:27.456 T:8304  WARNING <general>: CRenderManager::Configure - timeout waiting for configure
2021-05-11 11:49:29.133 T:3304     INFO <general>: DX::DeviceResources::ResizeBuffers: 10 bit swapchain is used with 6 flip discard buffers and HDR output
2021-05-11 11:49:30.642 T:3304     INFO <general>: VideoPlayer: OnLostDisplay received
2021-05-11 11:49:30.642 T:3304     INFO <general>: VideoPlayer: OnResetDisplay received
2021-05-11 11:49:30.642 T:6016     INFO <general>: CActiveAESink::OpenSink - initialize sink
2021-05-11 11:49:30.650 T:6016     INFO <general>: CAESinkWASAPI::InitializeExclusive: Format is Supported - will attempt to Initialize
2021-05-11 11:49:30.664 T:6016     INFO <general>: CAESinkWASAPI::InitializeExclusive: WASAPI Exclusive Mode Sink Initialized using: AE_FMT_S16NE, 192000, 8
2021-05-11 11:49:31.489 T:5416  WARNING <general>: CVideoPlayerAudio: Display reset has occurred, checking for passthrough
2021-05-11 11:49:31.491 T:3304     INFO <general>: DX::DeviceResources::SetHdrMetaData: RP 0.680 0.320 | GP 0.265 0.690 | BP 0.150 0.060 | WP 0.313 0.329 | Max ML 4000 | min ML 0.0050 | Max CLL 349 | Max FALL 86
2021-05-11 11:49:31.491 T:3304     INFO <general>: CRendererBase::ProcessHDR: Switching to HDR rendering
2021-05-11 11:49:31.491 T:3304  WARNING <general>: CRendererHQ::CheckVideoParameters: chosen scaling method 15 is not supported by renderer
2021-05-11 11:49:31.533 T:3304     INFO <general>: Loading skin file: VideoFullScreen.xml, load type: KEEP_IN_MEMORY
```

**Comment by thexai** [comment]
New test build based on Matrix v19.1:
http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210513-74ebe750-test-Matrix-x64.exe

**Comment by maisun** [comment]
> New test build based on Matrix v19.1:
> http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210513-74ebe750-test-Matrix-x64.exe

I can confirm this version also works for me - best thing happened in 2021  :-)

**Comment by SanderSp** [comment]
I'm subscribed to this topic since the beginning. I'd just like to thank you guys for continuing to bring attention to this problem. I'm still on a very old version of Kodi because of this issue and I will try the new version this weekend! Very happy!

**Comment by jhanda1** [comment]
> New test build based on Matrix v19.1:
> http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210513-74ebe750-test-Matrix-x64.exe

omg it works!!!! Thanks so much!  This is huge for me

**Comment by SanderSp** [comment]
I'm not familiar with Kodi development. Does this mean the fix will be implemented in future final versions of the software? In other words, can I update now knowing it'll keep working in the future when I get automatic updates, or should I wait a bit to see if it really gets implemented so I don't have to do my whole Kodi setup twice?

**Comment by maisun** [comment]
@thexai, I'd like to report a potential issues, some of my test videos which don't come with sound track stops playing with your fix. The videos I use are downloaded from: https://www.avsforum.com/threads/hdr10-test-patterns-set.2943380/

**Comment by thexai** [comment]
@maisun, I can not reproduce this. 

Are you saying that some files cause Kodi crash and _same_ files with regular v19.1 and _same_ settings not produces crash?

You can install regular v19.1 over test build and vice-versa to be sure.

**Comment by fm2511** [comment]
> New test build based on Matrix v19.1:
> http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210513-74ebe750-test-Matrix-x64.exe

It's not working for me. Using HTPC win 10 64bit with HD 530 video. 
Only manually toggle off/on audio passthrough works.

**Comment by thexai** [comment]
@fm2511, Try this build:
http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210517-01185241-test-Matrix-x64.exe

It uses audio device config changed events instead of display reset events. Should be more precise and work in all cases.

**Comment by fm2511** [comment]
@thexai, the last build works fine, no delay nedeed.
Great job, thanks!

**Comment by iz8mbw** [comment]
Hi @thexai 
any backport of this on Kodi 19 Matrix? I use Kodi 19.1 on Android TV.
Thanks!

**Comment by maisun** [comment]
> Hi @thexai
> any backport of this on Kodi 19 Matrix? I use Kodi 19.1 on Android TV (Nvidia Shield).
> Thanks!

As I understand this only affect Windows.

**Comment by fm2511** [comment]
Hi @thexai 
I just tested the new version 19.2, but the audio passthrough seems to work for me only after adding 2 or more seconds in 'Delay after change of refresh rate' option.

The last build that I tested worked fine, no delay nedeed (http://mirrors.kodi.tv/test-builds/windows/win64/KodiSetup-20210517-01185241-test-Matrix-x64.exe)

Is something changed in the final version 19.2?

**Comment by thexai** [comment]
The merged fix in Matrix is somewhat "less aggressive" than other variants tested during development. We believe that the other option could cause side effects and this is safer and more compatible.

Anyway, setting the Delay after change refresh rate to 2 - 3 seconds (depending on the hardware) is the right thing to do because it is the time it takes TV and AVR to change the video mode (especially if also is switching SDR - HDR modes). This also prevents playback from starting without the image being visible.

**Comment by deskbump** [comment]
I was having the same issue in Kodi build 20.1 when mediacenter was resuming from Sleep. 
For me the issue was fixed by setting the Audio / Output Configuration to Optimized.

**Comment by chengtat** [comment]
I have the same issue with 5800x 3090 kodi 20.1, only N driver 516.40 can solve this problem
